### PR TITLE
Fixes for signac 2

### PR DESCRIPTION
**Pull request by cbkerr**
## Description

I did a pass for common fixes going to signac 2.
In addition, query arguments passed in JSON must be lowercase, so I fixed this in the documentation.

## Motivation and Context

Resolves: #192 

## Checklist:

- [X] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-docs/blob/main/CONTRIBUTING.md).
- [X] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-docs/blob/main/ContributorAgreement.md).
- [X] My code follows the [code style guideline](https://github.com/glotzerlab/signac-docs/blob/main/CONTRIBUTING.md#code-style) of this project.

### CODE REVIEW

**Review by b-butler — CHANGES_REQUESTED** [review]
We need to revert the Boolean changes and mention that the command line interface expects JSON while the Python interface requires valid Python.

It is confusing for users, but it's better to explain it this way to avoid errors on one interface or the other.

File: docs/source/query.rst [hunk 1]
```diff
@@ -168,8 +168,8 @@ Exists Operator
 ---------------
 
 If you want to check for the existence of a specific key but do not care about its actual value, use the ``$exists``-operator.
-For example, the expression ``{'p': {'$exists': True}}`` will return all documents that *have a key p* regardless of its value.
-Likewise, using ``False`` as argument would return all documents that have no key with the given name.
+For example, the expression ``{'p': {'$exists': true}}`` will return all documents that *have a key p* regardless of its value.
+Likewise, using ``false`` as argument would return all documents that have no key with the given name.
```
[the remark this answers is not in the document]
Comment (cbkerr) [reply docs/source/query.rst:172]: I think the confusion is what on this page refers to CLI vs Python.

File: docs/source/query.rst [hunk 2]
```diff
@@ -247,24 +247,25 @@ Simplified Syntax on the Command Line
 It is possible to use search expressions directly on the command line, for example in combination with the ``$ signac find`` command.
 In this case filter arguments are expected to be provided as valid JSON expressions.
 For simple filters, you can use a simplified syntax instead of writing JSON.
-For example, instead of ``{'p': 2}``, you can simply type ``p 2``.
+For example, instead of ``{'p': 2}``, you can type ``p 2``.
 
 A simplified expression consists of key-value pairs in alternation.
 The first argument will then be interpreted as the first key, the second argument as the first value, the third argument as the second key, and so on.
-If you provide an odd number of arguments, the last value will default to ``{'$exists': True}``.
+If you provide an odd number of arguments, the last value will default to ``{'$exists': true}``.
```
[the remark this answers is not in the document]
Comment (cbkerr) [reply docs/source/query.rst:254]: This code is in the section "Simplified Syntax on the Comand Line" so I expected it to work on the CLI

**Review by cbkerr — COMMENTED** [review]
I agree we should clarify that the python interface still needs proper True and False and clarify when we are in each one

The table at the end is only for CLI. I think we need to add a column to the table for Python and CLI

Here is some terminal output showing what works on the CLI.

Setup:
```bash
$ signac init
Initialized project.
$ signac job -c '{"a":1}'
42b7b4f2921788ea14dac5566e6f06d0
$ signac job -c '{"b":2}'
a1ad2ae30a5e8cc1ecb9663235105278
```
Note what it interprets arguments as:
```bash
$ signac find a 1
Interpreted filter arguments as '{"a": 1}'.
42b7b4f2921788ea14dac5566e6f06d0
```

Exactly pasted as in documentation now
```bash
$ signac find {'a': {'$exists': True}}
Interpreted filter arguments as '{"{a:": "{$exists:", "True}}": {"$exists": true}}'.
```

Try to amend with double quotes on the outside
```bash
$ signac find "{'a': {'$exists': True}}"
Failed to parse query argument. Ensure that '{'a': {'': True}}' is valid JSON!
Error: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)
```

Swap double and single quotes
```bash
$ signac find '{"a": {"$exists": True}}'
Failed to parse query argument. Ensure that '{"a": {"$exists": True}}' is valid JSON!
Error: Expecting value: line 1 column 19 (char 18)
```

Use JSON true instead of Python True
```bash
(py310) Quince:signac-query Corwin$ signac find '{"a": {"$exists": true}}'
42b7b4f2921788ea14dac5566e6f06d0
```

This matches what signac spits out "interpreted filter argments as"
```bash
$ signac find a
Interpreted filter arguments as '{"a": {"$exists": true}}'.
42b7b4f2921788ea14dac5566e6f06d0
```

**Review by melodyyzh — APPROVED** [review]
Talked to @cbkerr in person, looks good to me

**Pull request merged**